### PR TITLE
Enhancing GitLab provider with filtering projects by pattern RegExp

### PR DESCRIPTION
**Pull request by acierto**
#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by acierto** [comment]
Hi @benjdlambert, I see that you are assigned as a reviewer.
The entire idea of this PR to get a bit more filtering to the GitLab provider.
Currently it is possible only to specify the group for that, but in most of the cases it's not enough.
Like for example in one of our company's Gitlab's we have no groups but a lot of projects.
It's possible to filter projects by users, namespaces, etc. 
What can bring really a lot of configuration options on backstage side. 
So my idea is to provide somehow close option as in GitLab Processor, where you can use a '*', 
but here it's a bit advanced option with a regexp.

File: .changeset/cool-months-tickle.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+'@backstage/plugin-catalog-backend-module-gitlab': minor
```
Comment (benjdlambert) [inline .changeset/cool-months-tickle.md:2]: Can be a patch release as the plugin is `<1.0.0`

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Just need a change to the changeset and I think it's looking good!

File: .changeset/cool-months-tickle.md (hunk 1 above)
Comment (acierto) [reply .changeset/cool-months-tickle.md]: Updated

**Review by benjdlambert — APPROVED** [review]
OK - let's go with this! :tada

**Pull request merged**